Source/DafnyCore/Verifier/Translator.ClassMembers.cs

```diff
@@ -1353,6 +1355,49 @@ public partial class Translator {
       builder.Add(Assert(tok, q, new PODesc.TraitFrame(m.WhatKind, true), kv));
     }
 
+    // Return a way to know if an assertion should be converted to an assumptions
```

Comment:
The notion of "before" and "after" here seems potentially confusing. For instance, if you were to say `assert {:only "before"} P;` in the `else` branch of an `if` statement, it would enable assertions in the other branch of the `if` statement. It's a simple notion, though, so perhaps the strangeness of it isn't fatal.

Reply:
Yeah I thought about that too. The advantage is that the notions are straightforward indeed for now. I primarily envision users to use `{:only}` by itself for a single proof, and if they really want, they can delimit without surprise which assertions should be verified using `{:only "before"}` and `{:only "after"}`. It's just that I had to give meaning to single instances of  `{:only "before"}` and  `{:only "after"}`.

I also thought of a more coarse-grained numerical notion of "level", which is one that @jtristan suggested one day. For example, `assert {:only 0}` will be the same as `assert {:only}`, whereas `assert {:only 1}` will define a verification interval on the surrounding block, `assert {:only 2}` on the block containing that block, etc. Because we traverse the program using a visitor pattern and nodes, this would be quite easy to implement.
If you like this idea, that would make three of us to like it so I would happily implement it as part of this PR.

Reply:
Yeah, that makes sense. And adding those would be backward-compatible, so no need to hold up this PR for it.